distributed/worker_state_machine.py

```diff
@@ -457,6 +482,39 @@ class GatherDepDoneEvent(StateMachineEvent):
     __slots__ = ()
 
 
+@dataclass
+class ComputeTaskEvent(StateMachineEvent):
+    key: str
+    who_has: dict[str, Collection[str]]
+    nbytes: dict[str, int]
+    priority: tuple[int, ...]
+    duration: float
+    run_spec: SerializedTask
+    resource_restrictions: dict[str, float]
+    actor: bool
+    annotations: dict
+    __slots__ = tuple(__annotations__)  # type: ignore
+
+    def __post_init__(self):
+        # Fixes after msgpack decode
+        if isinstance(self.priority, list):
+            self.priority = tuple(self.priority)
+
+        if isinstance(self.run_spec, dict):
+            self.run_spec = SerializedTask(**self.run_spec)
+        elif not isinstance(self.run_spec, SerializedTask):
+            self.run_spec = SerializedTask(task=self.run_spec)
```

Comment:
A bit nitpicky, feel free to ignore. But it might be nice to use [init-only variables](https://docs.python.org/3/library/dataclasses.html#init-only-variables) for `function`, `args`, `kwargs`, and `task`, make `run_spec` a non-init field, and go back to this when constructing the message:
```python
    if isinstance(ts.run_spec, dict):
        msg.update(ts.run_spec)
    else:
        msg["task"] = ts.run_spec
```
Just a little more explicitness and type checking around what has to be passed.

Reply:
The reason why I packed it this way was to avoid replicating all the (fairly convoluted) annotations in the SerializedTask class. The SerializedTask class is messy to begin with - given more time, I'd redesign it thoroughly.